tests/integration/pilot/cross_revision_test.go

```diff
@@ -0,0 +1,74 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/tests/integration/pilot/common"
+)
+
+// TestRevisionTraffic checks that traffic between all revisions specified works.
+// This is similar to TestMultiVersionRevision, except it doesn't actually install the version of Istio under test.
+// This allows a conformance-style tests against existing installations in the cluster.
+// The test is completely skipped if ISTIO_TEST_EXTRA_REVISIONS is not set
+// To run, add each revision to test. eg `ISTIO_TEST_EXTRA_REVISIONS=canary,my-rev`.
+func TestRevisionTraffic(t *testing.T) {
+	rawExtraRevs, f := os.LookupEnv("ISTIO_TEST_EXTRA_REVISIONS")
```

Comment:
Maybe we can consider using the `IstioVersions` framework flag instead of adding a new envvar?  (Not sure how we're planning on running this test, if it needs its own make target for custom setup anyways `IstioVersions` might make sense, otherwise maybe not).

Reply:
isn't that used to install those versions? the intent of this is to run against existing revisions in the cluster for cases when we cannot install from the framework. Likely paired with kube.deploy=false

Reply:
No at the moment `IstioVersions` does not install versions, it just lets the test framework know they're there.

Reply:
Ah I see. I don't think it will work for this use case as it relies on Istio 1.10+

Reply:
But I did try it out and it works great for 1.10+